tests/src/test/scala/quasiquotes/QuasiquoteSuite.scala

```diff
@@ -306,4 +392,286 @@ class QuasiquoteSuite extends FunSuite {
     val q"$x" = q"42"
     assert(x.show[Syntax] === "42")
   }
+
+  test("arg\"$name = $expr\"") {
+    val name = q"x"
+    val expr = q"foo"
+    assert(arg"$name = $expr".show[Syntax] === "x = foo")
+  }
+
+  test("arg\"$expr: _*\"") {
+    val expr = q"foo"
+    assert(arg"$expr: _*".show[Syntax] === "foo: _*")
+  }
+
+  test("arg\"$expr\"") {
+    val expr = q"foo"
+    assert(q"$expr".show[Syntax] === "foo")
+  }
+
+  test("t\"$ref.$tname\"") {
+    val ref = q"X"
+    val tname = t"Y"
+    assert(t"$ref.$tname".show[Structure] === "Type.Select(Term.Name(\"X\"), Type.Name(\"Y\"))")
+  }
+
+  test("t\"$tpe#$tname\"") {
+    val tpe = t"X"
+    val tname = t"Y"
+    assert(t"$tpe#$tname".show[Syntax] === "X#Y")
+  }
+
+  test("t\"$ref.type\"") {
+    val ref = q"X"
+    assert(t"$ref.type".show[Syntax] === "X.type")
+  }
+
+  test("t\"$tpe[..$tpes]") {
+    val tpe = t"X"
+    val tpes = List(t"Y", t"Z")
+    assert(t"$tpe[..$tpes]".show[Syntax] === "X[Y, Z]")
+  }
+
+  test("t\"$tpe $tname $tpe\"") {
+    val tpe1 = t"X"
+    val tname = t"Y"
+    val tpe2 = t"Z"
+    assert(t"$tpe1 $tname $tpe2".show[Syntax] === "X Y Z")
+  }
+
+  test("t\"(..$atpes) => $tpe\"") {
+    val atpes = List(t"X", t"Y")
+    val tpe = t"Z"
+    assert(t"(..$atpes) => $tpe".show[Syntax] === "(X, Y) => Z")
+  }
```

Comment:
Here they are `Type.Name`, isn't it `Type.Arg`? They both have `t` StringContext for creating.